src/journal/catalog.c

```diff
@@ -94,25 +94,85 @@ const struct hash_ops catalog_hash_ops = {
         .compare = catalog_compare_func
 };
 
+static bool next_header(const char **s) {
```

Comment:
This seems mostly duplicitous of find_header(), perhaps get them to reuse this piece?